azurerm/internal/services/containers/kubernetes_addons.go

```diff
@@ -175,6 +175,10 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
 								ConflictsWith: []string{"addon_profile.0.ingress_application_gateway.0.subnet_cidr", "addon_profile.0.ingress_application_gateway.0.subnet_id"},
 								ValidateFunc:  applicationGatewayValidate.ApplicationGatewayID,
 							},
+							"gateway_name": {
+								Type:     schema.TypeString,
+								Optional: true,
```

Comment:
This needs just a little bit of basic validation to prevent some wonky scenarios from happening. 
```
ValidateFunc: validation.StringIsNotEmpty,
```

Reply:
Done, thanks!!